scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowDoFns.scala

```diff
@@ -75,40 +73,70 @@ private[this] abstract class PredictDoFn[T, V, M <: Model[_]](
 
     c.output(result)
   }
-
-  override def createResource(): ConcurrentMap[String, M] =
-    new ConcurrentHashMap[String, M]()
-
 }
 
-private[tensorflow] class SavedBundlePredictDoFn[T, V](
+private[tensorflow] abstract class SavedBundlePredictDoFn[T, V](
   uri: String,
-  options: TensorFlowModel.Options,
-  fetchOp: Seq[String],
-  inFn: T => Map[String, Tensor[_]],
-  outFn: (T, Map[String, Tensor[_]]) => V
-) extends PredictDoFn[T, V, TensorFlowModel](fetchOp, inFn, outFn) {
+  options: TensorFlowModel.Options
+) extends PredictDoFn[T, V, TensorFlowModel] {
   @transient private lazy val log = LoggerFactory.getLogger(this.getClass)
 
-  override def withRunner(f: Session#Runner => V): V = {
-    val model = getResource
-      .computeIfAbsent(
-        uri,
-        new Function[String, TensorFlowModel] {
-          override def apply(t: String): TensorFlowModel =
-            Models
-              .tensorFlow(uri, options)
-              .get(Duration.ofDays(Integer.MAX_VALUE))
-        }
-      )
+  @Setup
+  def setup(): Unit =
+    model = Models
```

Comment:
@martinbomio hm, not sure about this;`DoFnWithResource` was used here to avoid loading the model per **instance**; each worker will create N instances of this `DoFn`. This might cause some memory issues if big models get loaded.

Reply:
hmm, ok makes sense. I wanted to simplify the resource logic a bit, I believe we are incorrectly using a Map as a resource, since the `DoFn` only support a single model. I will bring back the `DoFnWithResource `